dockstore-webservice/src/main/java/io/dockstore/webservice/resources/MetadataResource.java

```diff
@@ -169,6 +177,33 @@ public String rssFeed() {
         }
     }
 
+    @GET
+    @Produces({ "text/plain", "application/json" })
+    @Path("/runner_dependencies")
+    @ApiOperation(value = "Returns the file containing runner dependencies", response = String.class)
+    public Response getRunnerDependencies(
+            @ApiParam(value = "The Dockstore client version") @QueryParam("client_version") String clientVersion,
+            @ApiParam(value = "Python version, only relevant for the cwltool runner") @DefaultValue("2") @QueryParam("python_version") String pythonVersion,
+            @ApiParam(value = "The tool runner", allowableValues = "cwltool") @DefaultValue("cwltool") @QueryParam("runner") String runner,
+            @ApiParam(value = "Response type", allowableValues = "json, text") @DefaultValue("text") @QueryParam("output") String output,
+            @Context ContainerRequestContext containerRequestContext) {
+        if (!("cwltool").equals(runner)) {
+            return Response.noContent().build();
+        }
+        boolean unwrap = !("json").equals(output);
+        String fileVersion = PipHelper.convertSemVerToAvailableVersion(clientVersion);
+        try {
+            String content = Resources.toString(this.getClass().getClassLoader()
+                    .getResource("requirements/" + fileVersion + "/requirements" + (pythonVersion.startsWith("3") ? "3" : "") + ".txt"), StandardCharsets.UTF_8);
+            Map<String, String> pipDepMap = PipHelper.convertPipRequirementsStringToMap(content);
+            return Response.status(Response.Status.OK).type(unwrap ? MediaType.TEXT_PLAIN : MediaType.APPLICATION_JSON)
+                    .entity(unwrap ? content : pipDepMap).build();
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Use LOG.error rather than a stack trace and throw a CustomWebApplicationException as with other end points, looks like error code 500 in this case.